db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SubqueryNode.java

```diff
@@ -220,6 +222,31 @@ public void init(
         this.leftOperand=(ValueNode)leftOperand;
     }
 
+    public void setProperties(Properties properties) throws StandardException {
+        if (properties == null)
+            return;
+
+        Enumeration e=properties.keys();
+        while(e.hasMoreElements()){
+            String key=(String)e.nextElement();
+            String value=(String)properties.get(key);
+
+            switch(key){
+                case "doNotFlatten":
+                    try {
+                        hintNotFlatten = Boolean.parseBoolean(StringUtil.SQLToUpperCase(value));
+                    } catch (Exception exception) {
+                        throw StandardException.newException(SQLState.LANG_INVALID_FORCED_NO_SUBQUERY_FLATTEN_VALUE, value);
+                    }
+                    break;
```

Comment:
This 'break' only jumps out of the switch part, I think it's better to jump out of the while loop also.

Reply:
Thanks @changli6 . Right now, subquery only supports one hint, in the future, if we have more hints to support, we don't want to break out the while loop.